### PR TITLE
VACMS-17473 Vet Center Spotlights should use URL.path if available or URI if not

### PR DESCRIPTION
**Pull request by eselkin**
## Summary

- Internal references in links have a URL path but other links do not always. If a URL path exists that should be the link used. If not, defer to the URI (most often it is the link URL). There may be edge cases where a URL path doesn't exist but I could not find them.
- Sitewide Facilities

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-cms/issues/17473

## Testing done

- Tested on local and tugboat
- Checked the links for spotlights on: http://d14d5336cf01fbca2683256477352442.review.vetsgov-internal/hilo-vet-center/
- CTAs appear as normal when still a URI and appear with URL path for node's location

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |   Screenshot 2024-03-18 at 12 28 36 PM     |    Screenshot 2024-03-18 at 12 29 39 PM   |
| Desktop |    Screenshot 2024-03-18 at 12 27 35 PM    |   Screenshot 2024-03-18 at 12 26 27 PM    |

## What areas of the site does it impact?

Affects the VetCenter and VBA Spotlight Featured content only.

## Acceptance criteria

- [x] When an internal link is used the link points to the externally-accessible URL
- [x] When a regular link is used the link points to the externally-accessible URL

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [x] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [x] Events are being sent to the appropriate logging solution
- [x] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [x] Did you login to a local build and verify all authenticated routes work as expected with a test user

## Requested Feedback

Check on Review Instance for Vet Center at Hilo:
http://d14d5336cf01fbca2683256477352442.review.vetsgov-internal/hilo-vet-center/

VBA check needs to be built locally because VBA facilities are not published on staging. However, the logic is the same for VBA and Vet Center for the spotlight

### CODE REVIEW

**Review by maxx1128 — COMMENTED** [review]
One small tweak, otherwise this looks good.

File: src/site/includes/vba_facilities/spotlight_content.liquid [hunk 1]
```diff
@@ -9,11 +9,11 @@
         </div>
     {% endif %}
     {% if entity.fieldCta.entity.fieldButtonLink %}
-        {% if entity.fieldCta.entity.fieldButtonLink.url %}
+        {% if entity.fieldCta.entity.fieldButtonLink.url.path %}
```
[the remark this answers is not in the document]
Comment (eselkin) [reply src/site/includes/vba_facilities/spotlight_content.liquid]: updated if

**Comment by laflannery** [comment]
@eselkin Is there a new link available for me to review this? The one in the testing steps above is not loading

**Comment by eselkin** [comment]
@laflannery the review-instance changed: http://d14d5336cf01fbca2683256477352442.review.vetsgov-internal/hilo-vet-center

**Comment by laflannery** [comment]
Looks good @eselkin, Thanks!

File: src/site/includes/vet_centers/featured_content.liquid [hunk 2]
```diff
@@ -10,8 +10,14 @@
...
-            <va-link active href="{{ entity.fieldCta.entity.fieldButtonLink.uri }}" text="{{ entity.fieldCta.entity.fieldButtonLabel }}"></va-link>
+        {% assign fieldButtonUrl = entity.fieldCta.entity.fieldButtonLink.url.path | hasCharacterOtherThanSpace %}
+        {% if fieldButtonUrl %}
+            {% assign fieldUrl = entity.fieldCta.entity.fieldButtonLink.url.path %}
+        {% elsif fieldButtonUri %}
+            {% assign fieldUrl = entity.fieldCta.entity.fieldButtonLink.uri %}
+        {% endif %}
+        {% if fieldUrl %}
+            <va-link active href="{{ fieldUrl }}" text="{{entity.fieldCta.entity.fieldButtonLabel}}"></va-link>
         {%endif%}
```
Comment (maxx1128) [inline src/site/includes/vet_centers/featured_content.liquid:21]: **Nitpick (non-blocking):** Spaces needed `endif`.

File: src/site/includes/vet_centers/featured_content.liquid [hunk 3]
```diff
@@ -10,8 +10,14 @@
     {% endif %}
     {% if entity.fieldCta.entity.fieldButtonLink %}
         {% assign fieldButtonUri = entity.fieldCta.entity.fieldButtonLink.uri | hasCharacterOtherThanSpace %}
-        {% if fieldButtonUri %}
-            <va-link active href="{{ entity.fieldCta.entity.fieldButtonLink.uri }}" text="{{ entity.fieldCta.entity.fieldButtonLabel }}"></va-link>
+        {% assign fieldButtonUrl = entity.fieldCta.entity.fieldButtonLink.url.path | hasCharacterOtherThanSpace %}
```
Comment (maxx1128) [inline src/site/includes/vet_centers/featured_content.liquid:13]: **Praise:** good use of this liquid filter.

File: src/site/includes/vba_facilities/spotlight_content.liquid [hunk 4]
```diff
@@ -8,16 +8,14 @@
...
-                    <va-link 
-                        href="{{ entity.fieldCta.entity.fieldButtonLink.url }}" 
-                        text="{{ entity.fieldCta.entity.fieldButtonLabel }}" 
-                    ></va-link>
-                </p>
-            </div>
-        {% endif %}
+    {% if entity.fieldCta.entity.fieldButtonLink.url.path %}
+        <div class="vads-u-padding-top--1">
+            <p>
```
Comment (maxx1128) [inline src/site/includes/vba_facilities/spotlight_content.liquid:13]: **Suggestion:** I think this `div` and `p` tag could be combined to `<p class="vads-u-padding-top--1">`

**Review by maxx1128 — CHANGES_REQUESTED** [review]

File: src/site/includes/vet_centers/featured_content.liquid (hunk 2 above)
Comment (eselkin) [reply src/site/includes/vet_centers/featured_content.liquid:21]: spaced

File: src/site/includes/vba_facilities/spotlight_content.liquid (hunk 4 above)
Comment (eselkin) [reply src/site/includes/vba_facilities/spotlight_content.liquid:13]: changed.

**Review by maxx1128 — APPROVED** [review]

**Pull request merged**